lang/java/compiler/src/main/velocity/org/apache/avro/compiler/specific/templates/java/classic/record.vm

```diff
@@ -129,6 +129,25 @@ public class ${this.mangle($schema.getName())}#if ($schema.isError()) extends or
   }
 #end
 
+#if ($this.hasLogicalTypeField($schema))
```

Comment:
I tried out this branch and it pretty much worked, but I believe the location of this block is problematic. It is hard to tell, but it is actually in a foreach loop, so if your schema has multiple fields a logical type, it will generate invalid Java. The fix is to move this down below the #end block on line 151. 

Here is a sample schema that can reproduce the error:
https://gist.github.com/ppearcy/39e40489ab709e58d9a3